pkg/doc/verifiable/credential.go

```diff
@@ -239,48 +244,89 @@ func WithDisabledCustomSchemaCheck() CredentialOpt {
 	}
 }
 
-// NewCredential creates an instance of Verifiable Credential by reading a JSON document from bytes
-func NewCredential(data []byte, opts ...CredentialOpt) (*Credential, error) {
+// WithCredentialBuilder defines custom credential builder which usually used when data model should be extended.
+func WithCredentialBuilder(cb CredentialBuilder) CredentialOpt {
+	return func(opts *credentialOpts) {
+		opts.cb = cb
+	}
+}
+
+func issuerMapping(data []byte, credential *Credential) (err error) {
+	issuerID, issuerName, err := issuerFromBytes(data)
+	if err != nil {
+		err = fmt.Errorf("JSON unmarshalling of Verifiable Credential bytes failed: %w", err)
+		return
+	}
+
+	credential.Issuer = Issuer{ID: issuerID, Name: issuerName}
+	return
+}
+
+// CredentialBuilder provides a way to extend Credential Data model.
+// It defines a factory method to create new Credential instance and
+// provides decoders needed to fill custom fields of Credential.
+type CredentialBuilder interface {
+	New() *Credential
```

Comment:
I think it is confusing to have a New() inside an interface.

Perhaps CreateCredential?

Reply:
And its method set seems disjointed. I suggest splitting them into two new options, perhaps with names `WithTemplate` and `WithDecoder`.

Reply:
I like the simplification @llorllale proposed!